sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/NetFrameworkVersion.cs

```diff
@@ -29,6 +29,8 @@ public enum NetFrameworkVersion
         // between .NET 4.0 (inclusive) and .NET 4.5.1 (inclusive)
         Between4And451,
         // after .NET 4.5.2 (inclusive)
-        After452
+        After452,
```

Comment:
Can you change this to `Is452`, because after 4.5.2, there was 4.6. And "After452" doesn't really represent the state, since the detection doesn't return `After452` for higher versions.

It also creates some usage issue in `XmlExternalEntityShouldNotBeParsed`, see my comment there.